### PR TITLE
Use S.originalRequest in Menu snippet

### LINKED ISSUE #1695: Menu.item evaluation in AJAX requests
https://github.com/lift/framework/blob/master/web/webkit/src/main/scala/net/liftweb/builtin/snippet/Menu.scala#L443

For AJAX requests, there is no such Loc. S.originalRequest.map(_.location) meaning that Menu.item will never be correctly evaluated, S.originalRequest.flatMap(_.location) would probably be better in the snippet. Tim Nelson tested this and it does indeed fix the problem.

Mailing list thread: https://groups.google.com/forum/#!topic/liftweb/z9L8p7gKO64

### PR DESCRIPTION
**Pull request by eltimn**
Fixes #1695

### CODE REVIEW

File: web/webkit/src/test/scala/net/liftweb/builtin/snippet/MenuSpec.scala [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+import http._
+import mockweb._
+import MockWeb._
+import mocks._
+import sitemap._
+import util.Helpers.{randomString, secureXML}
+
+/**
+ * System under specification for Menu.
+ */
```
Comment (Shadowfiend) [inline web/webkit/src/test/scala/net/liftweb/builtin/snippet/MenuSpec.scala]: This seems like a pretty redundant comment? Might be auto-generated though, dunno.

Comment (eltimn) [inline web/webkit/src/test/scala/net/liftweb/builtin/snippet/MenuSpec.scala]: I agree. Those are in a lot if not all of our tests. Not sure where they came from.

**Comment by Shadowfiend** [comment]
Took a quick look at trying to figure out how to properly mock `S.originalRequest`, was also unsuccessful. Probably worth keeping in the back of our minds to see if we can make that clearer in the future, but for now this looks good.

Looks like the travis build blew up in a variety of ways, going to rerun it and see if it's branch-related or not.

**Comment by eltimn** [comment]
@Shadowfiend the tests fail due to setting SiteMap being set. I'm almost done with a solution.

**Comment by Shadowfiend** [comment]
Got it!

**Comment by eltimn** [comment]
I had to duplicate some of the mongo changes to get some tests passing.

**Comment by Shadowfiend** [comment]
This looks good to me; can probably ask for feedback from the original reporter as well once we merge!

**Pull request merged**